enacts/wat_bal/maproom_monit.py

```diff
@@ -492,23 +553,45 @@ def wat_bal_plots(
             error_msg="Please ensure all input boxes are filled for the calculation to run."
         )
 
+    #Save actual start of ts2
+    ts2_start = ts2["T"][0]
+
+    #Find corresponding day closer to ts
     p_d2 = calc.sel_day_and_month(
         precip["T"], int(planting2_day), calc.strftimeb2int(planting2_month)
     )
     p_d2 = p_d2.where(
         abs(ts["T"][0] - p_d2) == abs(ts["T"][0] - p_d2).min(), drop=True
     ).squeeze(drop=True).rename("p_d2")
+    #Assign ts-contemporary dates to ts2
     ts2 = ts2.assign_coords({"T": pd.date_range(datetime.datetime(
         p_d2.dt.year.values, p_d2.dt.month.values, p_d2.dt.day.values
     ), periods=ts2["T"].size)})
-
+    #Align ts and ts2 so that they have same size (otherwise Scatter goes bananas)
     ts, ts2 = xr.align(ts, ts2, join="outer")
+    #Recreate real dates of ts2 after alignment in ts contemporary dates
+    #to feed to customdata to hovertemplate
+    ts2_customdata = calc.sel_day_and_month(
+        precip["T"], ts2["T"][0].dt.day, ts2["T"][0].dt.month
+    )
+    ts2_customdata = ts2_customdata.where(
+        abs(ts2_start - ts2_customdata) == abs(ts2_start - ts2_customdata).min(),
+        drop=True,
+    ).squeeze(drop=True)
+    ts2_customdata = pd.date_range(datetime.datetime(
+        ts2_customdata.dt.year.values,
```

Comment:
`datetime.datetime` can't do anything with arrays,  so i guess you're counting on the `squeeze` to turn `ts2_customdata` into a scalar? Is that guaranteed to work? What if there are two entries for the same date? Too unrealistic to be worth checking for?

I don't understand how `pd.date_range` is being used. You specify `start` and `periods`, but neither `freq` nor `end`, so how does it know how wide to make the intervals?

Reply:
correct, if no squeeze, error is returned:

TypeError: only integer scalar arrays can be converted to a scalar index

where could return 2 values if the user picked a second planting day equidistant from 2 first planting dates of consecutive years (e.g. something like 1 Jul for the second p d and 1 Jan for the first -- probably not quite that but you get the idea). So it's possible even though the spirit of the Maproom is to compare with what-if considerations such as what if I had planted a couple weeks earlier/later, but in any case looking at the same season of interest thus not going 6 months away. But as it stands the Maproom allows it to happen.

[pd.date_range](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.date_range.html) default freq is "D"

Reply:
Since it's possible for there to be more than one entry, let's explicitly select the zeroth one, rather than crashing if we ever hit that case.

Reply:
done